lib/ansible/modules/cloud/amazon/ecs_attribute.py

```diff
@@ -0,0 +1,310 @@
+    attributes:
+        description:
+            - List of attributes.
+        required: true
+        contains:
+            name:
+                description:
+                    - The name of the attribute. Up to 128 letters (uppercase and lowercase),
+                      numbers, hyphens, underscores, and periods are allowed.
+                required: true
+            value:
+                description:
+                    - The value of the attribute. Up to 128 letters (uppercase and lowercase),
+                      numbers, hyphens, underscores, periods, at signs (@), forward slashes, colons,
+                      and spaces are allowed.
+                required: false
+    ec2_instance_id:
+        description:
+            - EC2 instance ID of ECS cluster container instance.
+        required: true
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Set attributes
+- ecs_attribute:
+    state: present
+    cluster: test-cluster
+    ec2_instance_id: "{{ ec2_id }}"
+    attributes:
+      - flavor: test
+      - migrated
+  delegate_to: localhost
+
+# Delete attributes
+- ecs_attribute:
+    state: absent
+    cluster: test-cluster
+    ec2_instance_id: "{{ ec2_id }}"
+    attributes:
+      - flavor: test
+      - migrated
+  delegate_to: localhost
+'''
+
+RETURN = '''
+attributes:
+    description: attributes
+    type: complex
+    contains:
+        cluster:
+            description: cluster name
+            type: string
+        ec2_instance_id:
+            description: ec2 instance id of ecs container instance
+            type: string
+        attributes:
+            description: list of attributes
+            type: list of complex
+            contains:
+                name:
+                    description: name of the attribute
+                    type: string
+                value:
+                    description: value of the attribute
+                    returned: if present
+                    type: string
+'''
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, EndpointConnectionError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info
+
+
+class EcsAttributes:
+    """Handles ECS Cluster Attribute"""
+
+    def __init__(self, module, attributes):
+        self.module = module
+        self.attributes = attributes if self._validate_attrs(attributes) else self._parse_attrs(attributes)
+
+    def __bool__(self):
+        return bool(self.attributes)
+
+    __nonzero__ = __bool__
+
+    def __iter__(self):
+        return iter(self.attributes)
+
+    @staticmethod
+    def _validate_attrs(attrs):
+        return all(('name', 'value') == tuple(attr.keys()) for attr in attrs)
```

Comment:
This isn't really working right for me since the dict keys are not necessarily in that order. Need some sorting.
Running this task:
```
      ecs_attribute:
        cluster: "{{ name }}"
        state: absent
        attributes:
          - flavor: test
          - migrated
        ec2_instance_id: "{{ ec2_id }}"
```
Results in:
```
fatal: [localhost]: FAILED! => {
    "changed": false,
    "failed": true,
    "invocation": {
        "module_args": {
            "attributes": [
                {
                    "flavor": "test"
                },
                "migrated"
            ],
            "aws_access_key": null,
            "aws_secret_key": null,
            "cluster": "testcluster",
            "ec2_instance_id": "i-07da5958fc74fbc",
            "ec2_url": null,
            "profile": "shertel",
            "region": "us-east-1",
            "security_token": null,
            "state": "absent",
            "validate_certs": true
        }
    },
    "msg": "Incorrect attribute format - {'value': 'test', 'name': 'flavor'}"
}

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1
```
I think this is happening because [line 255 ](https://github.com/ansible/ansible/pull/20618/files#diff-bf8ee38d299d949473d0a2221a344250R255) is calling EcsAttributes for the second time, after those attributes were already parsed once. So this error doesn't seem to happen when initially creating attributes, only when updating or removing already existing ones.